Build failure with D3D12 code with mingw-w64 headers earlier than from 2024-06-16

### Tested versions

- Not reproducible on b97110cd30
- Reproducible on e3550cb20f

### System information

Windows 10

### Issue description

When building with not-bleeding-edge mingw-w64 headers, presumably any versions earlier than https://github.com/mingw-w64/mingw-w64/commit/41d29b87719543cd55bd39c935e2dd8a67ff68eb (in my case, the headers from [llvm-mingw-20240606](https://github.com/mstorsjo/llvm-mingw/releases/tag/20240606)), I get the following error:

```
In file included from servers\display_server.cpp:42:
In file included from ./drivers/d3d12/rendering_context_driver_d3d12.h:38:
In file included from ./drivers/d3d12/rendering_device_driver_d3d12.h:55:
In file included from thirdparty\directx_headers\include\directx/d3dx12.h:11:
In file included from thirdparty\directx_headers\include\directx/d3d12.h:26:
In file included from D:/dev/toolchain/llvm-mingw/llvm-mingw-20240606-ucrt-x86_64/include/rpc.h:16:
In file included from D:/dev/toolchain/llvm-mingw/llvm-mingw-20240606-ucrt-x86_64/include/windows.h:97:
In file included from D:/dev/toolchain/llvm-mingw/llvm-mingw-20240606-ucrt-x86_64/include/winscard.h:10:
In file included from D:/dev/toolchain/llvm-mingw/llvm-mingw-20240606-ucrt-x86_64/include/wtypes.h:8:
D:/dev/toolchain/llvm-mingw/llvm-mingw-20240606-ucrt-x86_64/include/rpcndr.h:15:2: error: incorrect <rpcndr.h> version. Use the header that matches with the MIDL compiler.
   15 | #error incorrect <rpcndr.h> version. Use the header that matches with the MIDL compiler.
      |  ^
1 error generated.
```

I suspect it to be caused by this change: https://github.com/godotengine/godot/pull/91172/files#diff-bed1d5fa123300327771a6b6219ef4fed87354d7aa92dfdd9e72cde434de524fR42

CC @clayjohn

### Steps to reproduce

Build using a mingw-w64 toolchain that is not using bleeding-edge mingw-w64 headers, with d3d12 enabled.

### Minimal reproduction project (MRP)

n/a

--- a/drivers/d3d12/SCsub
+++ b/drivers/d3d12/SCsub
@@ -136,7 +136,6 @@ if env.msvc:
     ]
 else:
     extra_defines += [
-        ("__REQUIRED_RPCNDR_H_VERSION__", 475),
         "HAVE_STRUCT_TIMESPEC",
     ]
 

--- a/drivers/d3d12/rendering_device_driver_d3d12.h
+++ b/drivers/d3d12/rendering_device_driver_d3d12.h
@@ -36,6 +36,11 @@
 #include "core/templates/self_list.h"
 #include "servers/rendering/rendering_device_driver.h"
 
+#ifndef _MSC_VER
+// Match current version used by MinGW, MSVC and Direct3D 12 headers use 500.
+#define __REQUIRED_RPCNDR_H_VERSION__ 475
+#endif
+
 #if defined(__GNUC__) && !defined(__clang__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wnon-virtual-dtor"

